.github/workflows/reusable-integration-tests-on-prem-nightly.yml

```diff
@@ -0,0 +1,93 @@
+name: "Integration tests (on-prem, nightly build)"
```

Comment:
Could we clarify that it's the nightly build of Neo4j that we run the tests against? Otherwise it might be a bit confusing since there is also an `integration-tests-on-prem` workflow. Wdyt?